chromatic/rainbows/actions/shift.py

```diff
@@ -1,23 +1,33 @@
 from ...imports import *
 
-__all__ = ['shift']
+__all__ = ["shift"]
 
-def shift(self, velocity=5*u.km/u.s):
+
+def shift(self, velocity=0 * u.km / u.s):
     """
-    Apply a relativistic longitudinal doppler effect to the wavelength array
+    Doppler shift the wavelengths of this Rainbow.
+    Positive velocities make wavelengths longer (redshift).
+    Negative velocities make wavelengths shorter (bluesfhit).
 
     Parameters
     ----------
-    velocity
-        the systemic velocity in question, in km/s
+    velocity : astropy.units.Quantity
+        the systemic velocity in question,
+        with units of velocity (for example, u.km/u.s)
     """
-    
+
+    # create a new copy of this rainbow
     new = self._create_copy()
-    
-    lightspeed = con.c.to('km/s') #speed of light in km/s
 
-    beta = velocity/lightspeed
-    new_wavelength = new.wavelength * np.sqrt( (1 - beta) / (1 + beta) )
-    new.wavelike['wavelength'] = new_wavelength
-    
+    # get the speed of light from astropy constants
+    lightspeed = con.c.to("km/s")  # speed of light in km/s
+
+    # calculate beta and make sure the units cancel
+    beta = (velocity / lightspeed).decompose()
```

Comment:
The `.decompose()` function will make darn sure that the units have canceled out.

Reply:
Oh great!